### PR TITLE
Fix device list update to not constantly resync

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/handlers/device.py [hunk 1]
```diff
@@ -318,61 +369,79 @@ def _incoming_device_list_update(self, origin, edu_content):
...
+        )
+
+        yield self._handle_device_updates(user_id)
+
+    @measure_func("_incoming_device_list_update")
+    @defer.inlineCallbacks
+    def _handle_device_updates(self, user_id):
+        "Actually handle pending updates."
+
         with (yield self._remote_edue_linearizer.queue(user_id)):
```
Comment (NegativeMjark) [inline synapse/handlers/device.py]: "edue"?

File: synapse/handlers/device.py [hunk 2]
```diff
@@ -318,61 +369,79 @@ def _incoming_device_list_update(self, origin, edu_content):
...
+        for _, stream_id, prev_ids, _ in updates:
+            if not prev_ids:
+                # We always do a resync if there are no previous IDs
+                defer.returnValue(True)
+
+            for prev_id in prev_ids:
+                if prev_id == extremity:
+                    continue
+                elif prev_id in seen_updates:
+                    continue
```
Comment (NegativeMjark) [inline synapse/handlers/device.py]: Could we remove the prev_id from the seen_updates here?

Comment (erikjohnston) [inline synapse/handlers/device.py]: Possibly, but I'd rather keep it simple and just keep the most recently seen stream_ids.

**Comment by erikjohnston** [comment]
@NegativeMjark PTAL

**Pull request merged**